### PR TITLE
chore: use single command to toggle input/output

### LINKED ISSUE #186: Toggling IO Mapping in default rendering template component, breaks the component
__Describe the Bug__

Toggling IO Mapping in default rendering template component, breaks the component

__Steps to Reproduce__

1. Have template with default rendering for IO Mapping
2. Remove all IO Mappings
3. Add IO Mappings
=> IO Mapping is now broken

Peek 2021-11-18 17-07

__Expected Behavior__

IO Mapping does not break

__Environment__

https://github.com/bpmn-io/bpmn-properties-panel/pull/161

### PR DESCRIPTION
**Pull request by barmac**
Closes #186

This will need to be tested via build artifacts:

- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/element-templates/camunda-modeler-element-templates-linux-x64.tar.gz
- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/element-templates/camunda-modeler-element-templates-mac.dmg
- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/element-templates/camunda-modeler-element-templates-mac.zip
- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/element-templates/camunda-modeler-element-templates-win-ia32.zip
- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/element-templates/camunda-modeler-element-templates-win-x64.zip

Build link: https://github.com/camunda/camunda-modeler/actions/runs/1480901812

### CODE REVIEW

File: src/provider/element-templates/properties/InputProperties.js [hunk 1]
```diff
@@ -133,90 +133,80 @@ function LocalVariableAssignment(props) {
...
+    updatedBusinessObject = inputOutput;
+
+    const inputParameter = createInputParameter(binding, value, bpmnFactory);
+    inputParameter.$parent = inputOutput;
 
-  commandStack.execute('properties-panel.multi-command-executor', commands);
+    update = { inputParameters: inputOutput.get('camunda:inputParameters').concat(inputParameter) };
+  }
+
+  modeling.updateModdleProperties(element, updatedBusinessObject, update);
```
Comment (pinussilvestrus) [inline src/provider/element-templates/properties/InputProperties.js:172]: ❤️

**Comment by barmac** [comment]
You can use this template to test this:

```json
{
    "$schema": "https://unpkg.com/@camunda/element-templates-json-schema@0.4.0/resources/schema.json",
    "name": "Default Rendering",
    "id": "com.camunda.example.DefaultRendering",
    "appliesTo": [
      "bpmn:ServiceTask"
    ],
    "properties": [
      {
        "label": "Input Parameter",
        "binding": {
          "type": "camunda:inputParameter",
          "name": "processVar"
        }
      },
      {
        "label": "Output Parameter",
        "binding": {
          "type": "camunda:outputParameter",
          "source": "outputParam"
        }
      }
    ]
  }
```

© @MaxTru

**Comment by MaxTru** [comment]
What a nice simplification isn't it :-)

**Comment by barmac** [comment]
I've added some coverage. The artifacts are ready.

File: src/provider/element-templates/properties/OutputProperties.js [hunk 2]
```diff
@@ -181,84 +181,74 @@ function AssignToProcessVariable(props) {
...
-    });
-  }
+    const extensionElements = createExtensionElements(businessObject, bpmnFactory),
+          inputOutput = createInputOutput(binding, value, bpmnFactory, extensionElements);
+    extensionElements.values.push(inputOutput);
 
-  // (3) add camunda:InputParameter
-  const outputParamter = createOutputParameter(binding, value, bpmnFactory);
+    update = { extensionElements };
+  } else if (!inputOutput) {
```
Comment (MaxTru) [inline src/provider/element-templates/properties/OutputProperties.js:205]: Can we not have a testCase for (1) extensionElements exists, but (2) no input/output?

**Comment by MaxTru** [comment]
Nice! Works!

I only have a minor testCoverage comment

File: src/provider/element-templates/properties/OutputProperties.js (hunk 2 above)
Comment (barmac) [reply src/provider/element-templates/properties/OutputProperties.js:205]: Sure, we can cover'em all

Comment (barmac) [reply src/provider/element-templates/properties/OutputProperties.js:205]: Done.

**Review by MaxTru — APPROVED** [review]

**Pull request merged**